components/common/FormFields.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { ChangeEventHandler } from "react"
+import {
+    Button,
+    Checkbox,
+    Flex,
+    FormControl,
+    FormErrorMessage,
+    FormLabel,
+    Input,
+} from "@chakra-ui/react"
+import styled from "@emotion/styled"
+import ErrorAlert from "./ErrorAlert"
+
+type FormInputFieldProps = {
+    label: string
+    placeholder: string
+    isRequired: boolean
+    isInvalid?: string
+    formErrorMessage?: string
+    onChange: ChangeEventHandler<HTMLInputElement>
+}
+
+export function FormInputField({
+    label,
+    placeholder,
+    isRequired,
+    isInvalid,
+    formErrorMessage,
+    onChange,
+}: FormInputFieldProps): JSX.Element {
+    return (
+        <FormControl isInvalid={!isInvalid} isRequired={isRequired}>
```

Comment:
We should definitely design the component API so that we set `isInvalid={isInvalid}` here.